usr/lib/pkcs11/common/mech_aes.c

```diff
@@ -3145,12 +3148,21 @@ CK_RV ckm_aes_wrap_format(STDLL_TokData_t *tokdata,
         len2 = AES_BLOCK_SIZE * ((len1 / AES_BLOCK_SIZE) + 1);
 
         if (length_only == FALSE) {
-            ptr = (CK_BYTE *) realloc(*data, len2);
+            /*
+             * Don't use realloc here, since the buffer contains a key and the
+             * old buffer needs to be cleansed before it is freed.
+             */
+            ptr = (CK_BYTE *)malloc(len2);
             if (!ptr) {
                 TRACE_ERROR("%s\n", ock_err(ERR_HOST_MEMORY));
                 return CKR_HOST_MEMORY;
             } else {
                 memset(ptr + len1, 0x0, (len2 - len1));
+                if (*data != NULL) {
```

Comment:
can you remove the above "else" i.e. no "else" after "if {... return}"?

Reply:
fixed